pkg/compactor/compactor.go

```diff
@@ -614,6 +616,8 @@ func (c *Compactor) CompactTable(ctx context.Context, tableName string, applyRet
 		// wait for lock to be released since we can't mark delete requests as processed without checking all the tables
 		select {
 		case <-lockWaiterChan:
+			// refresh index list cache since the compaction would have changed the index files in the object store
+			sc.indexStorageClient.RefreshIndexTableNamesCache(ctx)
```

Comment:
we refresh the table [here](https://github.com/grafana/loki/blob/05c4b77ed02df5682d35db6582058c9e2764558c/pkg/compactor/table.go#L119), is this still required when compacting a table?

Reply:
ahh, right, this one is not required. Will remove it.